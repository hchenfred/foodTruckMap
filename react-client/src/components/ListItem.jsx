import React, { Component } from 'react';

class ListItem extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    this.props.truck.marker.setAnimation(google.maps.Animation.BOUNCE);
    setTimeout(() => {
      this.props.truck.marker.setAnimation(null);
     }, 1000);
  }

  render() {
    const {applicant, address, dayshours, fooditems} = this.props.truck;
    return (
      <div onClick={this.handleClick} className="card">
        <h5 className="card-title">
          {applicant}
        </h5>  
        <p><i className="fa fa-calendar" aria-hidden="true"></i> {dayshours}</p>
        <p><i className="fa fa-cutlery" aria-hidden="true"></i> {fooditems}</p>
        <p><i className="fa fa-map-marker" aria-hidden="true"></i> {address}</p>
      </div>
    );
  }
};

export default ListItem;