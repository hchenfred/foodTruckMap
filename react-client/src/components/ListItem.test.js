import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListItem from './ListItem';

test('ListItem component should render as expected', () => {
  const truck = {
    applicant: 'Munch A Bunch',
    address: '100 Brannan St',
    dayshours: 'Mo-Fr:10AM-11AM',
    fooditems: 'good food!',
    cnn: '123456'
  };
  const component = shallow(<ListItem truck={truck} key={'123456'}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});