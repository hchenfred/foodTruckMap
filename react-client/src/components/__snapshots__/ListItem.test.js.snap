// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should render as expected 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <h5
    className="card-title"
  >
    Munch A Bunch
  </h5>
  <p>
    <i
      aria-hidden="true"
      className="fa fa-calendar"
    />
     
    Mo-Fr:10AM-11AM
  </p>
  <p>
    <i
      aria-hidden="true"
      className="fa fa-cutlery"
    />
     
    good food!
  </p>
  <p>
    <i
      aria-hidden="true"
      className="fa fa-map-marker"
    />
     
    100 Brannan St
  </p>
</div>
`;
